R/dribble.R

```diff
@@ -0,0 +1,49 @@
+dribble <- function() {
+  structure(
+    tibble::tibble(
+      name = character(),
+      id = character(),
+      drive_file = list()
+    ),
+    class = c("dribble", "tbl_df", "tbl", "data.frame")
+  )
+}
+
+drive_id <- function(x) {
+  stopifnot(is.character(x))
+  structure(x, class = "drive_id")
+}
+
+as.dribble <- function(x, ...) UseMethod("as.dribble")
+
+as.dribble.dribble <- function(x, ...) x
+
+as.dribble.character <- function(x, ...) {
+  structure(
+    drive_path(x),
+    class = c("dribble", "tbl_df", "tbl", "data.frame")
+  )
+}
```

Comment:
All these explicit calls to `structure()` here will go away once functions like `drive_path()`, `drive_get()` etc. are brought into line and are returning dribble.